translate/tracer.go

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package translate
+
+// Identifier saves size and offset of the value which the variable has.
+type Identifier struct {
+	Offset uint
+	Size   uint
+}
+
+// MemoryTable is used to know the location of the memory
+type MemoryTable struct {
+	IdentifierMap map[string]Identifier
```

Comment:
what is key(id) of `IdentifierMap`?
I think id will be stored in bytecode instead of offset and size. am i right?

Reply:
The key is a variable's name. ID is not stored in bytecode. Offset and size would be stored in bytecode. For storing offset and size, we should know what is offset or size of the value. MemoryTable is used to know them.